File: user-service/src/main/java/socialapp/userservice/model/entity/BaseEntity.java
package socialapp.userservice.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@MappedSuperclass
@Getter
@Setter
public class BaseEntity {

    @JsonIgnore
    private LocalDateTime createdTime;
    @JsonIgnore
    private LocalDateTime lastModifiedTime;

    @PrePersist
    void prePersist(){
        this.createdTime = LocalDateTime.now();
    }

    @PreUpdate
    void preUpdate(){
        this.lastModifiedTime = LocalDateTime.now();
    }
}

File: user-service/src/main/kotlin/socialapp/ktuserservice/model/entity/BaseEntity.kt
package socialapp.ktuserservice.model.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.MappedSuperclass
import jakarta.persistence.PrePersist
import jakarta.persistence.PreUpdate
import org.springframework.data.annotation.Transient
import java.time.LocalDateTime

@MappedSuperclass
open class BaseEntity {
    @JsonIgnore
    @Transient
    var createdTime: LocalDateTime? = null
    @JsonIgnore
    @Transient
    var lastModifiedTime: LocalDateTime? = null

    @PrePersist
    fun prePersist() {
        this.createdTime = LocalDateTime.now()
    }

    @PreUpdate
    fun preUpdate() {
        this.lastModifiedTime = LocalDateTime.now()
    }
}